src/test/java/org/openstreetmap/atlas/streaming/resource/FileTest.java

```diff
@@ -0,0 +1,600 @@
+package org.openstreetmap.atlas.streaming.resource;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.streaming.compression.Compressor;
+import org.openstreetmap.atlas.streaming.compression.Decompressor;
+import org.openstreetmap.atlas.utilities.collections.Sets;
+
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+
+/**
+ * @author lcram
+ */
+public class FileTest
+{
+    @Rule
+    public final ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void testChild()
+    {
+        try (FileSystem filesystem = Jimfs.newFileSystem(Configuration.osX()))
```

Comment:
Not familiar with Jimfs... Does that mean this test wont work or will work differently on other platforms?

Reply:
No - it only affects the way the filesystem behaves under the various file operations. See the Javadoc comment [here](https://github.com/google/jimfs/blob/c934a4b71f8bc396107e40cfc7bfad7c1d45359e/jimfs/src/main/java/com/google/common/jimfs/Configuration.java#L56) as well as the explanations for the various configurations - [Unix configuration](https://github.com/google/jimfs/blob/c934a4b71f8bc396107e40cfc7bfad7c1d45359e/jimfs/src/main/java/com/google/common/jimfs/Configuration.java#L84), [osX configuration](https://github.com/google/jimfs/blob/c934a4b71f8bc396107e40cfc7bfad7c1d45359e/jimfs/src/main/java/com/google/common/jimfs/Configuration.java#L132), etc.

Reply:
I just picked `osX` for no particular reason. Could have just as well used the `unix` configuration for the purposes of these unit tests.